plugin/src/test/java/io/jenkins/plugins/casc/vault/VaultTestUtil.java

```diff
@@ -0,0 +1,76 @@
+package io.jenkins.plugins.casc.vault;
+
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container;
+import org.testcontainers.vault.VaultContainer;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class VaultTestUtil {
+    private final static Logger LOGGER = Logger.getLogger(VaultTestUtil.class.getName());
+
+    public static final String VAULT_DOCKER_IMAGE = "vault:1.0.3";
+    public static final String VAULT_ROOT_TOKEN = "root-token";
+    public static final String VAULT_USER = "admin";
+    public static final String VAULT_URL = "http://localhost:8200";
+    public static final String VAULT_PW = "admin";
+    public static final String VAULT_POLCY= "io/jenkins/plugins/casc/vault/vaultTest_adminPolicy.hcl";
+    public static final String VAULT_PATH_V1 = "kv-v1/admin";
+    public static final String VAULT_PATH_V2 = "kv-v2/admin";
+
+    private static Container.ExecResult runCommand(VaultContainer container, final String... command) throws IOException, InterruptedException {
+        LOGGER.log(Level.FINE, String.join(" ", command));
+        final Container.ExecResult result = container.execInContainer(command);
+        final String out = result.getStdout();
+        final String err = result.getStderr();
+        if (out != null && !out.isEmpty()) {
+            LOGGER.log(Level.FINE, result.getStdout());
+        }
+        if (err != null && !err.isEmpty()) {
+            LOGGER.log(Level.WARNING, result.getStderr());
+        }
+        return result;
+    }
+
+    public static boolean isWindowsNode() {
+        return System.getProperty( "os.name" ).startsWith( "Windows" );
```

Comment:
I would not trust os.name :sweat_smile:

```
public static boolean isWindows() {
  return File.pathSeparatorChar == ';';
}
```

Perhaps I would prefer if the test actually checked if `JENKINS_URL` is equal to `https://ci.jenkins.io` cause I would like to be able to run the test locally :smile:

Reply:
Oh, perhaps it would be better to check `TestEnvironment.dockerApiAtLeast`, no docker API would be reported therefore you could AssumeTrue Docker available :)

https://github.com/testcontainers/testcontainers-java/blob/c58a6f10670a43c31f50586502a0722a449ad2e6/core/src/main/java/org/testcontainers/utility/TestEnvironment.java#L14

Reply:
Very good idea! Thank you! 
I added it :+1:

Reply:
It seems that `TestEnvironment.dockerApiAtLeast` also needs docker to be installed xD

```
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:78)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:105)
	at org.testcontainers.DockerClientFactory.getActiveApiVersion(DockerClientFactory.java:288)
	at org.testcontainers.utility.TestEnvironment.dockerApiAtLeast(TestEnvironment.java:16)
	at io.jenkins.plugins.casc.vault.VaultTestUtil.hasDockerDaemon(VaultTestUtil.java:42)
	at io.jenkins.plugins.casc.vault.VaultTestUtil.createVaultContainer(VaultTestUtil.java:46)
	at io.jenkins.plugins.casc.vault.KV1WithUserTest.<clinit>(KV1WithUserTest.java:21)

```

Reply:
I would add a try/catch block around this now - if this Exception is caught, return `false. Maybe not the nicest way to rely on a catch here, but it is easy and would do the job